### PR TITLE
Add E2E test for exemplars

### LINKED ISSUE #4014: Add E2E test for exemplars API
Now the exemplars API is added and we already instrumented Thanos with exemplars in https://github.com/thanos-io/thanos/pull/3977. It is time to add an E2E test case for it.

AC:
~~1. Add a Jaeger all in one service in the E2E test as it is required for tracing~~
2. Update the options to start Thanos component with tracing configs
3. Call the exemplars API from thanos querier and check the results

For the request and response of the exemplars API, please check https://prometheus.io/docs/prometheus/latest/querying/api/#querying-exemplars

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

Fixes #4014 

## Changes

1. Use Builder pattern for creating Thanos Querier in the E2E test
2. Add some test cases for exemplars API

### CODE REVIEW

**Comment by yeya24** [comment]
Blocked by https://github.com/thanos-io/thanos/pull/4223. Will update the test after that pr is merged.

File: test/e2e/e2ethanos/services.go
Comment (kakkoyun) [inline test/e2e/e2ethanos/services.go:127]: :+1:

**Review by kakkoyun — APPROVED** [review]
LGTM :100:

**Review by onprem — APPROVED** [review]
LGTM!

**Comment by yeya24** [comment]
Looks like CI doesn't pass. WIll fix it in next pr.

**Comment by onprem** [comment]
> Looks like CI doesn't pass. WIll fix it in next pr.

The CI passed for this PR though, am I missing something?

**Pull request merged**
